File: backend/models/tramite.py
from pydantic import BaseModel
from typing import List, Optional

class Tramite(BaseModel):
    id: str
    nombre: str
    descripcion: Optional[str] = None
    costo: Optional[str] = "Gratuito"
    departamento: Optional[str] = None
    es_en_linea: Optional[bool] = True
    documentos_requeridos: Optional[List[str]] = []

File: src/types/tramite.ts
export interface ITramite {
  id: string;
  nombre: string;
  descripcion: string;
  costo: string;
  departamento: string;
  esEnLinea: boolean;
  documentosRequeridos: string[];
}

export interface ITimeSlot {
  hora: string;
  disponible: boolean;
}

export interface IAppointment {
  id?: string;
  tramiteId: string;
  fecha: string;
  hora: string;
  estado: "pendiente" | "confirmado" | "cancelado";
}

export interface IUploadedFile {
  nombre: string;
  file: File;
  estado: "subiendo" | "completado" | "error";
  progreso: number;
  url?: string;
}
